### DISCUSSION TITLE
Extension no longer working even after selecting free trial or entering payment info

### DISCUSSION THREAD

**Discussion by ZacTyAdams** [community/community#19105] [Copilot Conversations] [answered]
I've been a long time technical preview user and Copilot is showing `Extension activation failed: "User not authorized"` for both the free trial options and even after selecting a plan. I've tried restarting Visual Studio Code but there doesn't appear to be any change.

**Comment by ghost** [comment]
Same

**Comment by melvincwng** [comment]
Hi @ZacTyAdams and @Leartiker, to resolve the issue, kindly raise a support ticket and contact [Github Support over here](https://support.github.com/)

Hope this helps.

**Reply by melvincwng** [reply]
@ZacTyAdams, if this issue is resolved, please kindly 'Mark as answer' so that others are aware of the solution.

Thank you 😊

**Reply by ZacTyAdams** [reply]
this did not resolve the issue, there is no area to post a Github Support ticket for Copilot

**Reply by benlindsay** [reply]
I'm wondering if what @melvincwng meant was [this link](https://support.github.com/request?tags=docs-generic). It looks like this:

![image](https://user-images.githubusercontent.com/9361152/175053635-7021c809-6534-4a7d-a2d0-bc072a6f2079.png)

and I chose "Account or Billing Issues" and started filling out a support request there. However, it seems like it would be more efficient to discuss the issues here in this discussion so everyone can see what the problems and solutions are and solve our own problems rather than GitHub support staff having to deal with each issue one at a time.

**Reply by ZacTyAdams** [reply]
Yeah he sent me a direct link to enter a ticket in a later response but I'll keep this thread posted as I go along.

I've tried uninstalling the extension and removing my GitHub account entirely but I think the problem is with their activation server that's handing out the auth tokens.

**Comment by ShinChven** [comment]
Same here.

**Comment by ZacTyAdams** [comment]
@melvincwng this issue is not resolved and that link has no area to raise a support ticket for Copilot.

**Reply by melvincwng** [reply]
@ZacTyAdams, you can raise a ticket to Github Support over [here](https://support.github.com/contact/account).

**Comment by Siceberg** [comment]
Same

**Comment by benlindsay** [comment]
I have the same issue. In gory detail:

I've been using copilot for a while, and now, no matter what order I try uninstalling github copilot, signing out of github on VSCode, resigning in to github on VSCode, reinstalling github copilot, and attempting to authorize github copilot, I keep going through this sequence:

1. This pops up in VSCode

![image.png](https://github.zendesk.com/attachments/token/3nMsQziYUEWpVjTlzUwMrbJUa/?name=image.png)

2. Signup button takes me to this page

![image.png](https://github.zendesk.com/attachments/token/Dw8LVkFAxwnMDggpBnR5fVRCb/?name=image.png)

3. "Continue without payment details" link takes me here

![image.png](https://github.zendesk.com/attachments/token/v7bYvoh0GI24qyANpjSxVjq62/?name=image.png)

4. Look like success

![image.png](https://github.zendesk.com/attachments/token/TSmyf1WiB66DD3M4v19lGmqGA/?name=image.png)

5. But VSCode still shows "User not authorized"

![image.png](https://github.zendesk.com/attachments/token/siGIvXHzXhGhtoqj32LI3JmQB/?name=image.png)

6. And Output Log looks like this:

![image.png](https://github.zendesk.com/attachments/token/bnaj8agPaQ8d6d4JshYVNBH4i/?name=image.png)

**Reply by benlindsay** [reply]
For the record, I also created support ticket #1678480 with essentially the same contents

**Reply by ZacTyAdams** [reply]
Exact same  issue and steps taken here (even entered my payment info). Seems to have been an issue with their activation server, I put in a ticket #1678455 but will share whatever they tell me.

**Reply by cyaconi** [reply]
Please share what they reply. I'm having the same issue!

**Reply by bailehang** [reply]
Please share what they reply. I'm having the same issue!

**Reply by hsenag** [reply]
Hi, can I just check if everyone having problems has tried restarting VSCode after signing up? I know some people have said they did but not sure if everyone has.

**Reply by benlindsay** [reply]
I did, and signed out of Github and restarted my computer, and got a new social security number (well maybe not the last thing)

**Reply by hsenag** [reply]
Hi @benlindsay thanks a lot for the detailed report. We're still trying to work out what's happened, we'll get back to you as soon as we can.

**Reply by ZacTyAdams** [reply]
@hsenag yes restarted and reinstalled the extension. Any update on this?

**Reply by hsenag** [reply]
@ZacTyAdams I'm afraid not yet. Some other diagnostic/retry steps we think might be interesting (while admitting that we are still rather confused/stabbing in the dark a bit):
 - Logout from GitHub in VSCode and then log back in (as Support already suggested)
 - If not too painful, try any of the non-VSCode editors ([JetBrains, Visual Studio, Neovim](https://docs.github.com/en/copilot)) just to report if they work or not, as they use a slightly different authentication route.

**Reply by benlindsay** [reply]
Here's the response I got from support:

> Hi Ben Lindsay,
> 
> Thanks for the report. I've asked the team for their input on this one.
> 
> In the meantime, is there any chance you have tried signing out of GitHub completely as part of your troubleshooting steps. For some strange reason, I myself had trouble with my test account but after logging out of VSCode and GitHub.com itself, I was able to activate Copilot using these steps:
> 
>     1. Sign out of GitHub on VSCode. In the bottom left corner of VSCode there should be a user icon that will allow you to sign out.
>     2. Sign out of GitHub.com
>     3. Navigate to https://github.com/settings/copilot
>     4. Sign back into GitHub.com
>     5. Click Enable GitHub Copilot
>     6. Click Continue without payment details.
>     7. Select your preferences and click Save and get started
>     8. Launch VSCode. If you have the extension already installed for your code editor, you should now see a pop up requesting you to Sign in to access GitHub Copilot. Click theSign in to GitHub` button.
>     9. You will see a pop up stating The extension 'GitHub Copilot' wants to sign in using GitHub.. Click Allow.
>     10. A browser window should open with an oauth URL and you can click on Open Visual Studio Code.app.
>     11. A new window will pop up in VSCode asking you to Allow an extension to open this URI? Click Open.
>     12. You will get another pop up starting You have not yet finished authorizing this extension to use GitHub. Would you like to keep trying? Click Yes.
>     13. A GitHub Authentication browser window will open stating You are signed in now and can close this page.
>     14. Open VSCode and start using Copilot.
> 
> I know its quite a number of steps but hopefully following that you can start using Copilot again.
> 
> If that doesn't work for you, please let us know. Details about where you are running into trouble, the steps you've taken and any links or screenshots are also very helpful.

I tried those steps and they didn't work for me. The one new thing here is signing into and out of GitHub.com in the browser. Even though that didn't work, I'm wondering if there's some funky caching business going on, so I think I'll try biting the bullet and clearing my cache and data on Firefox in addition to those 14 steps and see if that does anything.

**Reply by ZacTyAdams** [reply]
@hsenag not that I understand the stack but I think the issue is on the authentication server side. Since it is returning a 403 code it is sending over correct credentials and pulling up the correct account, it's just not seeing the entitlement on the account. Is it possible the entitlement on the GitHub account from the technical preview is not being updated to the new paid entitlement?

**Reply by hsenag** [reply]
@ZacTyAdams I'd also love to understand that - is the client somehow buggy and sending something wrong to the server (despite all the resetting, logging out and in etc), so the server reports you're unauthorized, or does the server wrongly think your account is unauthorized? We're improving our internal tooling to try to get more info on that at the moment.

**Reply by ZacTyAdams** [reply]
@hsenag if you are using something like sailpoint or okta for IAM, this just looks like an entitlement/role/permission issue. You maybe able to quickly confirm by looking at our Identity and verifying we have the correct permission assigned and that our legacy one has been removed.

**Reply by benlindsay** [reply]
One thing I'm noticing is that if I navigate to https://github.com/settings/copilot, no matter how many times I have clicked that button and gone through the authorization process, it still shows up like this:

![image](https://user-images.githubusercontent.com/9361152/175324455-570c3383-5f5d-49bc-aa4a-107c765c01aa.png)

So this also makes me think that it isn't some problem with the client or networking between my computer and GitHub.com. It looks like an internal GitHub problem.

@ZacTyAdams, @Siceberg, @cyaconi, @bailehang, @Leartiker, and others, is this the same for you?

**Reply by ZacTyAdams** [reply]
Yeah that's what I was saying above, they need to see what entitlement is not getting assigned to our account.

**Reply by ZacTyAdams** [reply]
@benlindsay @hsenag Yeah I'm seeing the same when I look at my settings 
![image](https://user-images.githubusercontent.com/6806769/175350935-ee0915d6-f076-4b81-9939-8800cd46ddf9.png)

**Reply by benlindsay** [reply]
Support "discovered a quirk that might have been preventing full authorization of some users' tokens, and we've accordingly given the system a nudge." After that, going through the signout and signin process worked! But weirdly, my copilot settings page still doesn't seem to register that I have copilot activated. Still looks just like @ZacTyAdams 's screenshot. But I'm happy to let that remain a mystery since copilot is working.

**Reply by ZacTyAdams** [reply]
> @hsenag if you are using something like sailpoint or okta for IAM, this just looks like an entitlement/role/permission issue. You maybe able to quickly confirm by looking at our Identity and verifying we have the correct permission assigned and that our legacy one has been removed.

I told this same thing to support and it sounds like that was the problem, RBAC is hard! Glad it's back up and running!

**Comment by ShinChven** [comment]
I solved this problem by contacting support team

https://support.github.com/tickets/personal/0

**Reply by benlindsay** [reply]
@ShinChven that's a personal ticket that only you can see. Can you copy+paste or summarize the steps that worked for you?

**Comment by Siceberg** [comment]
I get support team reply,
> Can you try to completely sign out of GitHub using the Accounts icon located at the bottom left of VS Code, restart VS Code, then go through the sign-on process again

But not working for me.

**Reply by ShinChven** [reply]
I suggest you to try IDEA copilot plugin (not asking you to replace VSCODE with IDEA, just to figure out if it's an account issue.).
If you account does not work on IDEA's copilot plugin either, then it's not a VSCODE extension problem.
My problem is a billing state problem, GitHub support team helped solved it at their end.
I actually saw `Extension activation failed: "User not authorized"` message while my account was in an unexpected billing state.

**Reply by Siceberg** [reply]
I think this is my account problem, because I get vs code notification: "Your Copilot experience is not fully configured, complete your setup." + "Copilort setting" Button, After clicking the button, open browser with URL: https://github.com/settings/copilot?editor=copilot_1.30.6165, this page shows "You have not enabled Copilot for your account. Get started with a 60-day free trial.", then click "Enable Github Copilot" button, go to URL:https://github.com/github-copilot/tp_signup, shows "Thank you for being part of the GitHub Copilot technical preview. You can continue to use GitHub Copilot for free until August 22, 2022.", at this time, reenter https://github.com/settings/copilot?editor=copilot_1.30.6165, this page still shows "You have not enabled Copilot for your account. Get started with a 60-day free trial.", Copilot still not work, even restart vs code again.

**Comment by ghost** [comment]
I was able to activate Copilot by restarting vscode and signing in and out in github from vscode.

**Comment by ZacTyAdams** [comment]
So far Support has suggested:

Can you sign out of GitHub using the Accounts icon located at the bottom left of VS Code, restart VS Code, then go through the sign-on process again, via the Accounts icon?

This has not resolved the problem for me at this time, the authentication server is still returning a 403 code.

**Comment by canuckjacq** [comment]
Hey all! I have something for you all to check. 

We've seen quite a few issues in Support where Copilot doesn't enable fully because of billing issues. 🤑  

This can be any number of things and not all of them are caused by user error, but one is reasonably common and usually fixable in your settings: 

⚠️ **Are you seeing a red banner at the top of GitHub when you are logged in?** ⚠️ 

If you had an old coupon or an old paid plan and your card is expired - whatever - and you haven't got around to [downgrading to Free](https://docs.github.com/en/billing/managing-billing-for-your-github-account/downgrading-your-github-subscription#downgrading-your-personal-accounts-subscription), do that. 

If you had a Student coupon expire and you are still a student, you can [reapply](https://education.github.com/pack/join). 

So, while you're waiting for a reply from support, please first [check your billing settings](https://github.com/settings/billing). If anything there is out of whack, then copilot will probably not enable. 

I hope that gets some folks unblocked! We're working on getting the rest sorted as fast as we can. ❤️

**Reply by ZacTyAdams** [reply]
I'm not sure what changed but support was able to get me squared away doing this (re-entering card info) and I think changing something with my permissions on the backend.

**Reply by benlindsay** [reply]
I didn't/don't have any billing issues, but support "discovered a quirk that might have been preventing full authorization of some users' tokens, and we've accordingly given the system a nudge." After that, going through the signout and signin process worked! But weirdly, at this page: https://github.com/settings/copilot, it still doesn't seem to register that I have copilot activated. Looks like this:

![image](https://user-images.githubusercontent.com/9361152/175454396-ed714430-c3b1-42e4-b5da-8a71266035fe.png)

But I'm happy to let that remain a mystery since copilot is working.

**Comment by Siceberg** [comment]
New replay from support team 
> Thanks so much for trying that! I can see that you successfully authenticated, generating a new token. We are experiencing a small number of users affected by this issue, and our engineers are working on a fix. While we don't yet have a specific ETA, we'll be sure to keep you updated with any news.

**Reply by Siceberg** [reply]
Hopefully it will be done by August 22nd

**Comment by ZacTyAdams** [comment, marked as answer]
The issue was resolved after support "discovered a quirk that might have been preventing full authorization of some users' tokens, and we've accordingly given the system a nudge." (thanks for sharing that @benlindsay)

**Comment by PreetM07** [comment]
I am facing this now! Can someone please tell me what is giving nudge to the system?
Should this be a valid answer?

**Reply by ZacTyAdams** [reply]
Yes
